src/libgap-api.c

```diff
@@ -42,46 +68,135 @@ Obj GAP_ValueGlobalVariable(const char * name)
     }
 }
 
-//
-// Evaluate a string of GAP commands
-//
-// To see an example of how to use this function
-// see tst/testlibgap/basic.c
-//
-Obj GAP_EvalString(const char * cmd)
+
+////
+//// arithmetic
+////
+
+int GAP_EQ(Obj a, Obj b)
 {
-    Obj instream;
-    Obj res;
-    Obj viewObjFunc, streamFunc;
+    return EQ(a, b);
+}
 
-    streamFunc = GAP_ValueGlobalVariable("InputTextString");
-    viewObjFunc = GAP_ValueGlobalVariable("ViewObj");
+int GAP_LT(Obj a, Obj b)
+{
+    return LT(a, b);
+}
 
-    instream = DoOperation1Args(streamFunc, MakeString(cmd));
-    res = READ_ALL_COMMANDS(instream, False, True, viewObjFunc);
-    return res;
+int GAP_IN(Obj a, Obj b)
+{
+    return IN(a, b);
+}
+
+Obj GAP_SUM(Obj a, Obj b)
+{
+    return SUM(a, b);
+}
+
+Obj GAP_DIFF(Obj a, Obj b)
+{
+    return DIFF(a, b);
+}
+
+Obj GAP_PROD(Obj a, Obj b)
+{
+    return PROD(a, b);
+}
+
+Obj GAP_QUO(Obj a, Obj b)
+{
+    return QUO(a, b);
+}
+
+Obj GAP_LQUO(Obj a, Obj b)
+{
+    return LQUO(a, b);
+}
+
+Obj GAP_POW(Obj a, Obj b)
+{
+    return POW(a, b);
+}
+
+Obj GAP_COMM(Obj a, Obj b)
+{
+    return COMM(a, b);
+}
+
+Obj GAP_MOD(Obj a, Obj b)
+{
+    return MOD(a, b);
+}
+
+
+////
+//// booleans
+////
+
+Obj GAP_True;
+Obj GAP_False;
+Obj GAP_Fail;
+
+
+////
+//// lists
+////
+
+int GAP_IsList(Obj obj)
+{
+    return obj && IS_LIST(obj);
+}
+
+Int GAP_LenList(Obj obj)
+{
+    return LEN_LIST(obj);
+}
+
+void GAP_AssList(Obj list, UInt pos, Obj val)
+{
+    if (val)
+        ASS_LIST(list, pos, val);
+    else
+        UNB_LIST(list, pos);
+}
+
+Obj GAP_ElmList(Obj list, UInt pos)
+{
+    return ELM0_LIST(list, pos);
```

Comment:
Note that we use ELM0_LIST here, not ELM_LIST, as that makes it easier to write code using it, w/o having to add range check and `IsBound` checks (indeed, one would have to export IsBound in the first place... still might want to do that later, but for now, it's not there, as I never had reason to use it in an interface myself).

BTW, I just realize that there either should be a check for `pos == 0`, or it should be documented what happens if one passes in 0. Same for `GAP_AssList`

Reply:
I agree we definately need to check for pos == 0 (I do that all the time when writing extensions). I'm not sure if ELM0_LIST should catch that, or it should be caught here (I'm tempted to say ELM0_LIST should catch it)

Reply:
Hrm... Elm0Plist, ElmPlist, ElmPlistDense, all check that the index does not exceed the length, but none of them (nor their counterparts for blists, strings, ranges). This might or might not be a relevant performance optimization...

As such, I will for now simply add that check in `libgap-api.c`. I am tempted to say that in this case, 0 should be returned, as for all other "out of bounds" accesses (note that if one accidentally passes a negative index to this function, it'll be treated as a big positive integer, which then leads to 0 being returned, too). Of course it'd be trivial to instead call `ErrorMayQuit`, if people think that's better.